rosbag2_cpp/src/rosbag2_cpp/writers/sequential_writer.cpp

```diff
@@ -249,11 +249,13 @@ void SequentialWriter::write(std::shared_ptr<rosbag2_storage::SerializedBagMessa
     storage_->write(converted_msg);
```

Comment:
@emersonknapp we discussed cleaning this up better, however, I think this initial 0 check is required cause the cache size is set to 0 in this scenario (cause max_cache_size is 0) and so we need to handle it separately. 

Let me know what you think.